internal/webhook/management_webhook.go

```diff
@@ -121,73 +126,50 @@ func checkComponentsRemoval(ctx context.Context, cl client.Client, oldMgmt, newM
 		return nil
 	}
 
-	release := new(kcmv1.Release)
-	if err := cl.Get(ctx, client.ObjectKey{Name: newMgmt.Spec.Release}, release); err != nil {
-		return fmt.Errorf("failed to get Release %s: %w", newMgmt.Spec.Release, err)
-	}
-
-	removedProvidersSet := make(map[string]struct{})
+	inUseProviders := make(map[string]bool)
```

Comment:
```suggestion
	inUseProviders := make(map[string]struct{})
```

before `go1.24rc2` struct is more optimal memory-wise